File: CS/XpoXtraReports/OrderDocument.cs
using System;
using System.Drawing;
using System.Collections;
using System.Collections.Generic;
using System.ComponentModel;
using DevExpress.XtraReports.UI;
using DevExpress.Xpo;

namespace XpoXtraReports
{
    //[Report("0001","Order document",2)]
    public partial class OrderDocument : XpoXtraReports.XpoBaseReport
    {
        public OrderDocument()
        {
            InitializeComponent();
        }

        private void xpCollection1_ResolveSession(object sender, DevExpress.Xpo.ResolveSessionEventArgs e) {
            if (e.Session == null)
                e.Session = new UnitOfWork();
        }

    }
}

File: VB/XpoXtraReports/OrderDocument.vb
Imports Microsoft.VisualBasic
Imports System
Imports System.Drawing
Imports System.Collections
Imports System.Collections.Generic
Imports System.ComponentModel
Imports DevExpress.XtraReports.UI
Imports DevExpress.Xpo

Namespace XpoXtraReports
	'[Report("0001","Order document",2)]
	Partial Public Class OrderDocument
		Inherits XpoXtraReports.XpoBaseReport
		Public Sub New()
			InitializeComponent()
		End Sub

		Private Sub xpCollection1_ResolveSession(ByVal sender As Object, ByVal e As DevExpress.Xpo.ResolveSessionEventArgs) Handles xpCollection1.ResolveSession
			If e.Session Is Nothing Then
				e.Session = New UnitOfWork()
			End If
		End Sub

	End Class
End Namespace
